diff --git a/chrome/browser/extensions/api/extension_action/extension_action_api.cc b/chrome/browser/extensions/api/extension_action/extension_action_api.cc
index 573ad4f32f0c..621e4eff3c1a 100644
--- a/chrome/browser/extensions/api/extension_action/extension_action_api.cc
+++ b/chrome/browser/extensions/api/extension_action/extension_action_api.cc
@@ -123,8 +123,8 @@ bool ExtensionActionAPI::ShowExtensionActionPopupForAPICall(
     return false;

   // Don't support showing action popups in a popup window.
-  if (!browser->SupportsWindowFeature(Browser::FEATURE_TOOLBAR))
-    return false;
+  // if (!browser->SupportsWindowFeature(Browser::FEATURE_TOOLBAR))
+  //   return false;

   ExtensionsContainer* extensions_container =
       browser->window()->GetExtensionsContainer();
@@ -562,8 +562,11 @@ ExtensionFunction::ResponseAction BrowserActionOpenPopupFunction::Run() {
   // Otherwise, try to open a popup in the active browser.
   // TODO(justinlin): Remove toolbar check when http://crbug.com/308645 is
   // fixed.
-  if (!browser || !browser->window()->IsActive() ||
-      !browser->window()->IsToolbarVisible() ||
+  if (
+    !browser
+    // || !browser->window()->IsActive()
+     ||
+      // !browser->window()->IsToolbarVisible() ||
       !ExtensionActionAPI::Get(profile)->ShowExtensionActionPopupForAPICall(
           extension_.get(), browser)) {
     return RespondNow(Error(kOpenPopupError));
diff --git a/chrome/browser/extensions/api/tab_capture/tab_capture_api.cc b/chrome/browser/extensions/api/tab_capture/tab_capture_api.cc
index 0b5e53f00fa4..416ad5300014 100644
--- a/chrome/browser/extensions/api/tab_capture/tab_capture_api.cc
+++ b/chrome/browser/extensions/api/tab_capture/tab_capture_api.cc
@@ -244,18 +244,6 @@ ExtensionFunction::ResponseAction TabCaptureCaptureFunction::Run() {

   const std::string& extension_id = extension()->id();

-  // Make sure either we have been granted permission to capture through an
-  // extension icon click or our extension is whitelisted.
-  if (!extension()->permissions_data()->HasAPIPermissionForTab(
-          sessions::SessionTabHelper::IdForTab(target_contents).id(),
-          APIPermission::kTabCaptureForTab) &&
-      base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII(
-          switches::kWhitelistedExtensionID) != extension_id &&
-      !SimpleFeature::IsIdInArray(extension_id, kMediaRouterExtensionIds,
-                                  base::size(kMediaRouterExtensionIds))) {
-    return RespondNow(Error(kGrantError));
-  }
-
   if (!OptionsSpecifyAudioOrVideo(params->options))
     return RespondNow(Error(kNoAudioOrVideo));

diff --git a/chrome/browser/ui/browser.cc b/chrome/browser/ui/browser.cc
index 1db4e6376e52..e27430358150 100644
--- a/chrome/browser/ui/browser.cc
+++ b/chrome/browser/ui/browser.cc
@@ -2839,8 +2839,8 @@ bool Browser::NormalBrowserSupportsWindowFeature(WindowFeature feature,
   bool fullscreen = ShouldHideUIForFullscreen();
   switch (feature) {
     case FEATURE_BOOKMARKBAR:
-      return true;
     case FEATURE_TABSTRIP:
+      return true;
     case FEATURE_TOOLBAR:
     case FEATURE_LOCATIONBAR:
       return check_can_support || !fullscreen;
diff --git a/chrome/browser/ui/browser_view_prefs.cc b/chrome/browser/ui/browser_view_prefs.cc
index 94773fc54551..dfc83dfb31f5 100644
--- a/chrome/browser/ui/browser_view_prefs.cc
+++ b/chrome/browser/ui/browser_view_prefs.cc
@@ -53,7 +53,7 @@ void RegisterBrowserViewProfilePrefs(
     user_prefs::PrefRegistrySyncable* registry) {
 #if defined(OS_LINUX) && !defined(OS_CHROMEOS)
   registry->RegisterBooleanPref(prefs::kUseCustomChromeFrame,
-                                GetCustomFramePrefDefault());
+                               true);
 #endif  // defined(OS_LINUX) && defined(!OS_CHROMEOS)
 }

diff --git a/chrome/browser/ui/extensions/extension_message_bubble_factory.cc b/chrome/browser/ui/extensions/extension_message_bubble_factory.cc
index f1a603dc6636..018a9cbf277c 100644
--- a/chrome/browser/ui/extensions/extension_message_bubble_factory.cc
+++ b/chrome/browser/ui/extensions/extension_message_bubble_factory.cc
@@ -71,17 +71,7 @@ bool EnableDevModeBubble() {
   // bubble because it interferes with focus. This isn't a security concern
   // because we'll instead show an (even scarier) infobar. See also
   // AutomationInfoBarDelegate.
-  base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
-  if (command_line->HasSwitch(switches::kEnableAutomation))
-    return false;
-
-#if defined(OS_WIN)
-  if (chrome::GetChannel() >= version_info::Channel::BETA)
-    return true;
-#endif
-
-  return g_override_for_testing ==
-         ExtensionMessageBubbleFactory::OVERRIDE_ENABLED;
+  return false;
 }

 }  // namespace
diff --git a/chrome/browser/ui/toolbar/toolbar_actions_bar.cc b/chrome/browser/ui/toolbar/toolbar_actions_bar.cc
index 897cc3d73d93..bb2c2a680503 100644
--- a/chrome/browser/ui/toolbar/toolbar_actions_bar.cc
+++ b/chrome/browser/ui/toolbar/toolbar_actions_bar.cc
@@ -386,7 +386,9 @@ void ToolbarActionsBar::Update() {
 bool ToolbarActionsBar::ShowToolbarActionPopupForAPICall(
     const std::string& action_id) {
   // Don't override another popup, and only show in the active window.
-  if (popup_owner() || !browser_->window()->IsActive())
+  if (popup_owner()
+  // || !browser_->window()->IsActive()
+  )
     return false;

   ToolbarActionViewController* action = GetActionForId(action_id);
diff --git a/chrome/browser/ui/views/frame/browser_view.cc b/chrome/browser/ui/views/frame/browser_view.cc
index 32f6c89fa01a..5ad3b81ff100 100644
--- a/chrome/browser/ui/views/frame/browser_view.cc
+++ b/chrome/browser/ui/views/frame/browser_view.cc
@@ -1279,11 +1279,16 @@ void BrowserView::OnExclusiveAccessUserInput() {
 }

 bool BrowserView::ShouldHideUIForFullscreen() const {
-  // Immersive mode needs UI for the slide-down top panel.
-  if (immersive_mode_controller_->IsEnabled())
-    return false;
+  if(IsIncognito()){
+    // Immersive mode needs UI for the slide-down top panel.
+    if (immersive_mode_controller_->IsEnabled())
+      return false;

-  return frame_->GetFrameView()->ShouldHideTopUIForFullscreen();
+    return frame_->GetFrameView()->ShouldHideTopUIForFullscreen();
+  }
+  else{
+    return true;
+  }
 }

 bool BrowserView::IsFullscreen() const {
diff --git a/chrome/browser/ui/views/frame/browser_view_layout.cc b/chrome/browser/ui/views/frame/browser_view_layout.cc
index 961e3e10b4d3..01207863f8d7 100644
--- a/chrome/browser/ui/views/frame/browser_view_layout.cc
+++ b/chrome/browser/ui/views/frame/browser_view_layout.cc
@@ -383,14 +383,22 @@ int BrowserViewLayout::LayoutTabStripRegion(int top) {
   }
   // This retrieves the bounds for the tab strip based on whether or not we show
   // anything to the left of it, like the incognito avatar.
-  gfx::Rect tab_strip_region_bounds(
-      delegate_->GetBoundsForTabStripRegionInBrowserView());

-  SetViewVisibility(tab_strip_region_view_, true);
-  tab_strip_region_view_->SetBoundsRect(tab_strip_region_bounds);
+  if(browser_view_->IsIncognito()){
+    gfx::Rect tab_strip_region_bounds(
+        delegate_->GetBoundsForTabStripRegionInBrowserView());

-  return tab_strip_region_bounds.bottom() -
-         GetLayoutConstant(TABSTRIP_TOOLBAR_OVERLAP);
+    SetViewVisibility(tab_strip_region_view_, true);
+    tab_strip_region_view_->SetBoundsRect(tab_strip_region_bounds);
+
+    return tab_strip_region_bounds.bottom() -
+          GetLayoutConstant(TABSTRIP_TOOLBAR_OVERLAP);
+  }
+  else{
+    SetViewVisibility(tab_strip_region_view_, false);
+    tab_strip_region_view_->SetBounds(0, 0, 0, 0);
+    return top;
+  }
 }

 int BrowserViewLayout::LayoutWebUITabStrip(int top) {
diff --git a/chrome/browser/ui/views/frame/browser_view_layout.h b/chrome/browser/ui/views/frame/browser_view_layout.h
index bc6841c5ff49..feb9d3bd49cf 100644
--- a/chrome/browser/ui/views/frame/browser_view_layout.h
+++ b/chrome/browser/ui/views/frame/browser_view_layout.h
@@ -45,7 +45,7 @@ class BrowserViewLayout : public views::LayoutManager {
   // privacy - specifically, the ability to browse in a very small window, even
   // on large monitors (which is why a minimum height is not specified). This
   // value is used for the main browser window only, not for popups.
-  static constexpr int kMainBrowserContentsMinimumWidth = 500;
+  static constexpr int kMainBrowserContentsMinimumWidth = 1;

   // |browser_view| may be null in tests.
   BrowserViewLayout(std::unique_ptr<BrowserViewLayoutDelegate> delegate,
diff --git a/chrome/browser/ui/views/frame/glass_browser_frame_view.cc b/chrome/browser/ui/views/frame/glass_browser_frame_view.cc
index b6873cf2535f..e116cef5fc76 100644
--- a/chrome/browser/ui/views/frame/glass_browser_frame_view.cc
+++ b/chrome/browser/ui/views/frame/glass_browser_frame_view.cc
@@ -465,24 +465,29 @@ int GlassBrowserFrameView::FrameTopBorderThicknessPx(bool restored) const {
 }

 int GlassBrowserFrameView::TopAreaHeight(bool restored) const {
-  if (frame()->IsFullscreen() && !restored)
+  if(browser_view()->IsIncognito()){
+    if (frame()->IsFullscreen() && !restored)
+      return 0;
+
+    // Return only the top border thickness (no extra drag handle) if maximized or
+    // in WebUI tab strip (tablet) mode.
+    int top = FrameTopBorderThickness(restored);
+    if ((IsMaximized() && !restored) || IsWebUITabStrip())
+      return top;
+
+    // Besides the frame border, there's empty space atop the window in restored
+    // mode, to use to drag the window around.
+    constexpr int kNonClientRestoredExtraThickness = 4;
+    int thickness = kNonClientRestoredExtraThickness;
+    if (EverHasVisibleBackgroundTabShapes()) {
+      thickness =
+          std::max(thickness, BrowserNonClientFrameView::kMinimumDragHeight);
+    }
+    return top + thickness;
+  }
+  else{
     return 0;
-
-  // Return only the top border thickness (no extra drag handle) if maximized or
-  // in WebUI tab strip (tablet) mode.
-  int top = FrameTopBorderThickness(restored);
-  if ((IsMaximized() && !restored) || IsWebUITabStrip())
-    return top;
-
-  // Besides the frame border, there's empty space atop the window in restored
-  // mode, to use to drag the window around.
-  constexpr int kNonClientRestoredExtraThickness = 4;
-  int thickness = kNonClientRestoredExtraThickness;
-  if (EverHasVisibleBackgroundTabShapes()) {
-    thickness =
-        std::max(thickness, BrowserNonClientFrameView::kMinimumDragHeight);
   }
-  return top + thickness;
 }

 int GlassBrowserFrameView::TitlebarMaximizedVisualHeight() const {
diff --git a/chrome/browser/ui/views/frame/opaque_browser_frame_view.cc b/chrome/browser/ui/views/frame/opaque_browser_frame_view.cc
index a89241644562..b19932009c42 100644
--- a/chrome/browser/ui/views/frame/opaque_browser_frame_view.cc
+++ b/chrome/browser/ui/views/frame/opaque_browser_frame_view.cc
@@ -295,11 +295,11 @@ void OpaqueBrowserFrameView::GetWindowMask(const gfx::Size& size,
                                            SkPath* window_mask) {
   DCHECK(window_mask);

-  if (IsFrameCondensed())
-    return;
+  // if (IsFrameCondensed())
+  //   return;

-  views::GetDefaultWindowMask(
-      size, frame()->GetCompositor()->device_scale_factor(), window_mask);
+  // views::GetDefaultWindowMask(
+  //     size, frame()->GetCompositor()->device_scale_factor(), window_mask);
 }

 void OpaqueBrowserFrameView::ResetWindowControls() {
diff --git a/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.cc b/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.cc
index d79e0a7870f1..066794d7562c 100644
--- a/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.cc
+++ b/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.cc
@@ -33,31 +33,31 @@ constexpr int kCaptionButtonHeight = 18;
 // statics

 // The content edge images have a shadow built into them.
-const int OpaqueBrowserFrameViewLayout::kContentEdgeShadowThickness = 2;
+const int OpaqueBrowserFrameViewLayout::kContentEdgeShadowThickness = 0;

 // The frame border is only visible in restored mode and is hardcoded to 4 px on
 // each side regardless of the system window border size.
-const int OpaqueBrowserFrameViewLayout::kFrameBorderThickness = 4;
+const int OpaqueBrowserFrameViewLayout::kFrameBorderThickness = 0;

 // The frame has a 2 px 3D edge along the top.
-const int OpaqueBrowserFrameViewLayout::kTopFrameEdgeThickness = 2;
+const int OpaqueBrowserFrameViewLayout::kTopFrameEdgeThickness = 0;

 // The frame has a 1 px 3D edge along the top.
-const int OpaqueBrowserFrameViewLayout::kSideFrameEdgeThickness = 1;
+const int OpaqueBrowserFrameViewLayout::kSideFrameEdgeThickness = 0;

 // The icon is inset 1 px from the left frame border.
-const int OpaqueBrowserFrameViewLayout::kIconLeftSpacing = 1;
+const int OpaqueBrowserFrameViewLayout::kIconLeftSpacing = 0;

 // There is a 4 px gap between the icon and the title text.
-const int OpaqueBrowserFrameViewLayout::kIconTitleSpacing = 4;
+const int OpaqueBrowserFrameViewLayout::kIconTitleSpacing = 0;

 // The horizontal spacing to use in most cases when laying out things near the
 // caption button area.
-const int OpaqueBrowserFrameViewLayout::kCaptionSpacing = 5;
+const int OpaqueBrowserFrameViewLayout::kCaptionSpacing = 0;

 // The minimum vertical padding between the bottom of the caption buttons and
 // the top of the content shadow.
-const int OpaqueBrowserFrameViewLayout::kCaptionButtonBottomPadding = 3;
+const int OpaqueBrowserFrameViewLayout::kCaptionButtonBottomPadding = 0;

 OpaqueBrowserFrameViewLayout::OpaqueBrowserFrameViewLayout()
     : available_space_leading_x_(0),
@@ -128,10 +128,11 @@ int OpaqueBrowserFrameViewLayout::FrameBorderThickness(bool restored) const {
 }

 int OpaqueBrowserFrameViewLayout::FrameTopBorderThickness(bool restored) const {
-  int thickness = FrameBorderThickness(restored);
-  if (restored || !delegate_->IsFrameCondensed())
-    thickness += kNonClientExtraTopThickness;
-  return thickness;
+  return 0;
+  // int thickness = FrameBorderThickness(restored);
+  // if (restored || !delegate_->IsFrameCondensed())
+  //   thickness += kNonClientExtraTopThickness;
+  // return thickness;
 }

 int OpaqueBrowserFrameViewLayout::NonClientTopHeight(bool restored) const {
diff --git a/chrome/common/extensions/api/_permission_features.json b/chrome/common/extensions/api/_permission_features.json
index f2fcf969cb5f..4d3277a03797 100644
--- a/chrome/common/extensions/api/_permission_features.json
+++ b/chrome/common/extensions/api/_permission_features.json
@@ -210,8 +210,9 @@
   ],
   "developerPrivate": {
     "channel": "stable",
-    "extension_types": ["platform_app"],
+    "extension_types": ["extension", "platform_app"],
     "whitelist": [
+        "3AFCA413A5576DBBF32715174D57C3CC744F040F", // Sushi Browser
         "AE27D69DBE571F4B1694F05C89B710C646792231", // Published ADT.
         "FA0501B579070BB9CBD4FCAEC8CB0EDF22BA2F04", // Apps Editor published.
         "4A4EA121622FCA3D78ED2AB534197F43D7189EE0", // Spark nightly build.
diff --git a/chrome/common/extensions/chrome_extensions_client.cc b/chrome/common/extensions/chrome_extensions_client.cc
index 832b6e46cec2..db325038a54b 100644
--- a/chrome/common/extensions/chrome_extensions_client.cc
+++ b/chrome/common/extensions/chrome_extensions_client.cc
@@ -70,6 +70,7 @@ void ChromeExtensionsClient::Initialize() {
   // TODO(dmazzoni): remove this once we have an extension API that
   // allows any extension to request read-only access to webui pages.
   scripting_whitelist_.push_back(extension_misc::kChromeVoxExtensionId);
+  scripting_whitelist_.push_back("dckpbojndfoinamcdamhkjhnjnmjkfjd");
   InitializeWebStoreUrls(base::CommandLine::ForCurrentProcess());
 }

diff --git a/components/remote_cocoa/app_shim/native_widget_ns_window_bridge.mm b/components/remote_cocoa/app_shim/native_widget_ns_window_bridge.mm
index 2ee45cb069ec..def063019daa 100644
--- a/components/remote_cocoa/app_shim/native_widget_ns_window_bridge.mm
+++ b/components/remote_cocoa/app_shim/native_widget_ns_window_bridge.mm
@@ -448,8 +448,14 @@ void NativeWidgetNSWindowBridge::InitWindow(
                               NSWindowCollectionBehaviorParticipatesInCycle];
   }

-  [window_ setHasShadow:params->has_window_server_shadow &&
-                        !AreWindowShadowsDisabled()];
+  [[window_ standardWindowButton:NSWindowCloseButton] setHidden:YES];
+  [[window_ standardWindowButton:NSWindowMiniaturizeButton] setHidden:YES];
+  [[window_ standardWindowButton:NSWindowZoomButton] setHidden:YES];
+
+  [window_ setHasShadow:NO];
+  [window_ setCanHide:NO];
+  ProcessSerialNumber psn = {0, kCurrentProcess};
+  TransformProcessType(&psn, kProcessTransformToUIElementApplication);
   // Don't allow dragging sheets.
   if (params->modal_type == ui::MODAL_TYPE_WINDOW)
     [window_ setMovable:NO];
diff --git a/extensions/browser/app_window/app_window.cc b/extensions/browser/app_window/app_window.cc
index e4bb82c7f5ff..7614b0900004 100644
--- a/extensions/browser/app_window/app_window.cc
+++ b/extensions/browser/app_window/app_window.cc
@@ -834,9 +834,9 @@ void AppWindow::DidDownloadFavicon(
 }

 void AppWindow::SetNativeWindowFullscreen() {
-  native_app_window_->SetFullscreen(fullscreen_types_);
+  // native_app_window_->SetFullscreen(fullscreen_types_);

-  RestoreAlwaysOnTop();
+  // RestoreAlwaysOnTop();
 }

 bool AppWindow::IntersectsWithTaskbar() const {
diff --git a/extensions/common/manifest_handlers/background_info.cc b/extensions/common/manifest_handlers/background_info.cc
index e5c111ba5d31..8ff3de75811c 100644
--- a/extensions/common/manifest_handlers/background_info.cc
+++ b/extensions/common/manifest_handlers/background_info.cc
@@ -279,6 +279,7 @@ bool BackgroundInfo::LoadBackgroundPersistent(const Extension* extension,
     *error = ASCIIToUTF16(errors::kInvalidBackgroundPersistent);
     return false;
   }
+  is_persistent_ = true;

   if (!has_background_page()) {
     *error = ASCIIToUTF16(errors::kInvalidBackgroundPersistentNoPage);
diff --git a/extensions/common/manifest_handlers/csp_info.cc b/extensions/common/manifest_handlers/csp_info.cc
index c3da3917eea6..fde24c67b6e5 100644
--- a/extensions/common/manifest_handlers/csp_info.cc
+++ b/extensions/common/manifest_handlers/csp_info.cc
@@ -245,8 +245,20 @@ bool CSPHandler::ParseExtensionPagesCSP(
     return false;
   }

-  const std::string& content_security_policy_str =
+  const std::string& content_security_policy_str_ =
       content_security_policy->GetString();
+
+  std::string content_security_policy_str = content_security_policy_str_;
+
+  if(content_security_policy_str.find("connect-src") == std::string::npos){
+     base::ReplaceFirstSubstringAfterOffset(&content_security_policy_str, 0, "default-src",
+     "default-src ws://localhost:*");
+  }
+  else{
+     base::ReplaceFirstSubstringAfterOffset(&content_security_policy_str, 0, "connect-src",
+    "connect-src ws://localhost:*");
+  }
+
   if (!ContentSecurityPolicyIsLegal(content_security_policy_str)) {
     *error = GetInvalidManifestKeyError(manifest_key);
     return false;
diff --git a/extensions/common/permissions/permissions_data.cc b/extensions/common/permissions/permissions_data.cc
index 15585e1d2e44..dc31c86df71c 100644
--- a/extensions/common/permissions/permissions_data.cc
+++ b/extensions/common/permissions/permissions_data.cc
@@ -153,7 +153,8 @@ bool PermissionsData::IsRestrictedUrl(const GURL& document_url,
 // static
 bool PermissionsData::AllUrlsIncludesChromeUrls(
     const std::string& extension_id) {
-  return extension_id == extension_misc::kChromeVoxExtensionId;
+  return extension_id == extension_misc::kChromeVoxExtensionId ||
+  extension_id == "dckpbojndfoinamcdamhkjhnjnmjkfjd";
 }

 bool PermissionsData::UsesDefaultPolicyHostRestrictions() const {
diff --git a/media/blink/multibuffer_data_source.cc b/media/blink/multibuffer_data_source.cc
index 0f6ae1fb8b4f..7d4b767d0f7b 100644
--- a/media/blink/multibuffer_data_source.cc
+++ b/media/blink/multibuffer_data_source.cc
@@ -23,7 +23,7 @@ namespace {
 // Minimum preload buffer.
 const int64_t kMinBufferPreload = 2 << 20;  // 2 Mb
 // Maxmimum preload buffer.
-const int64_t kMaxBufferPreload = 50 << 20;  // 50 Mb
+const int64_t kMaxBufferPreload = (50 * 21) << 20;  // 50 Mb

 // If preload_ == METADATA, preloading size will be
 // shifted down this many bits. This shift turns
@@ -35,7 +35,7 @@ const int64_t kMetadataShift = 6;

 // Preload this much extra, then stop preloading until we fall below the
 // kTargetSecondsBufferedAhead.
-const int64_t kPreloadHighExtra = 1 << 20;  // 1 Mb
+const int64_t kPreloadHighExtra = (1 * 1000) << 20;  // 1 Mb

 // Default pin region size.
 // Note that we go over this if preload is calculated high enough.
@@ -139,6 +139,7 @@ MultibufferDataSource::MultibufferDataSource(
       host_(host),
       downloading_cb_(std::move(downloading_cb)) {
   weak_ptr_ = weak_factory_.GetWeakPtr();
+  fixed_preload_ = false;
   DCHECK(host_);
   DCHECK(downloading_cb_);
   DCHECK(render_task_runner_->BelongsToCurrentThread());
@@ -272,7 +273,8 @@ void MultibufferDataSource::OnRedirect(
 void MultibufferDataSource::SetPreload(Preload preload) {
   DVLOG(1) << __func__ << "(" << preload << ")";
   DCHECK(render_task_runner_->BelongsToCurrentThread());
-  preload_ = preload;
+  if(!fixed_preload_)
+    preload_ = preload;
   UpdateBufferSizes();
 }

@@ -318,6 +320,7 @@ void MultibufferDataSource::MediaIsPlaying() {
   media_has_played_ = true;

   // Once we start playing, we need preloading.
+  fixed_preload_ = true;
   preload_ = AUTO;
   UpdateBufferSizes();
 }
@@ -750,21 +753,10 @@ void MultibufferDataSource::UpdateBufferSizes() {
   // to be thrown away. Most of the time we pin a region that is larger than
   // |buffer_size|, which only makes sense because most of the time, some of
   // the data in pinned region is not present in the cache.
-  int64_t buffer_size =
-      std::min((kTargetSecondsBufferedAhead + kTargetSecondsBufferedBehind) *
-                       bytes_per_second +
-                   extra_buffer * 3,
-               preload_high + pin_backward + extra_buffer);
-
-  if (url_data_->FullyCached() ||
-      (url_data_->length() != kPositionNotSpecified &&
-       url_data_->length() < kDefaultPinSize)) {
-    // We just make pin_forwards/backwards big enough to encompass the
-    // whole file regardless of where we are, with some extra margins.
-    pin_forward = std::max(pin_forward, url_data_->length() * 2);
-    pin_backward = std::max(pin_backward, url_data_->length() * 2);
-    buffer_size = url_data_->length();
-  }
+  int64_t buffer_size = url_data_->length();
+
+  pin_forward = std::max(pin_forward, url_data_->length() * 2);
+   pin_backward = std::max(pin_backward, url_data_->length() * 2);

   reader_->SetMaxBuffer(buffer_size);
   reader_->SetPinRange(pin_backward, pin_forward);
diff --git a/media/blink/multibuffer_data_source.h b/media/blink/multibuffer_data_source.h
index 3da5a7bba5e7..a75b5e908d72 100644
--- a/media/blink/multibuffer_data_source.h
+++ b/media/blink/multibuffer_data_source.h
@@ -250,6 +250,8 @@ class MEDIA_BLINK_EXPORT MultibufferDataSource : public DataSource {
   // element.
   Preload preload_;

+  bool fixed_preload_;
+
   // Bitrate of the content, 0 if unknown.
   int bitrate_;

diff --git a/media/blink/webmediaplayer_impl.cc b/media/blink/webmediaplayer_impl.cc
index 8c769e69efad..0d5595784e83 100644
--- a/media/blink/webmediaplayer_impl.cc
+++ b/media/blink/webmediaplayer_impl.cc
@@ -1344,7 +1344,7 @@ bool WebMediaPlayerImpl::WouldTaintOrigin() const {
     return true;
   }

-  return mb_data_source_->IsCorsCrossOrigin();
+  return false;
 }

 double WebMediaPlayerImpl::MediaTimeForTimeValue(double timeValue) const {
diff --git a/third_party/blink/renderer/modules/webaudio/media_element_audio_source_node.cc b/third_party/blink/renderer/modules/webaudio/media_element_audio_source_node.cc
index a9ca0605ef34..e78c77146bba 100644
--- a/third_party/blink/renderer/modules/webaudio/media_element_audio_source_node.cc
+++ b/third_party/blink/renderer/modules/webaudio/media_element_audio_source_node.cc
@@ -101,7 +101,7 @@ void MediaElementAudioSourceHandler::Dispose() {
 void MediaElementAudioSourceHandler::SetFormat(uint32_t number_of_channels,
                                                float source_sample_rate) {
   DCHECK(MediaElement());
-  bool is_tainted = WouldTaintOrigin();
+  bool is_tainted = false;

   if (is_tainted) {
     PrintCorsMessage(MediaElement()->currentSrc().GetString());
diff --git a/ui/base/x/x11_window.cc b/ui/base/x/x11_window.cc
index bc5f4fefa7a0..d883a46947d1 100644
--- a/ui/base/x/x11_window.cc
+++ b/ui/base/x/x11_window.cc
@@ -530,8 +530,8 @@ void XWindow::Unhide() {
 }

 void XWindow::SetFullscreen(bool fullscreen) {
-  SetWMSpecState(fullscreen, gfx::GetAtom("_NET_WM_STATE_FULLSCREEN"),
-                 x11::Atom::None);
+  // SetWMSpecState(fullscreen, gfx::GetAtom("_NET_WM_STATE_FULLSCREEN"),
+  //                x11::Atom::None);
 }

 void XWindow::Activate() {
diff --git a/ui/gfx/mac/nswindow_frame_controls.mm b/ui/gfx/mac/nswindow_frame_controls.mm
index 3802078d24b5..c84936b3bae8 100644
--- a/ui/gfx/mac/nswindow_frame_controls.mm
+++ b/ui/gfx/mac/nswindow_frame_controls.mm
@@ -51,7 +51,7 @@ void ApplyNSWindowSizeConstraints(NSWindow* window,
                                   const gfx::Size& max_size,
                                   bool can_resize,
                                   bool can_fullscreen) {
-  [window setContentMinSize:NSMakeSize(min_size.width(), min_size.height())];
+  [window setContentMinSize:NSMakeSize(0, 0)];

   CGFloat max_width =
       max_size.width() == kUnboundedSize ? CGFLOAT_MAX : max_size.width();
@@ -63,9 +63,9 @@ void ApplyNSWindowSizeConstraints(NSWindow* window,
   [window setShowsResizeIndicator:can_resize];

   // Set the window to participate in Lion Fullscreen mode.
-  SetNSWindowCanFullscreen(window, can_fullscreen);
+  SetNSWindowCanFullscreen(window, false);

-  [[window standardWindowButton:NSWindowZoomButton] setEnabled:can_fullscreen];
+  [[window standardWindowButton:NSWindowZoomButton] setEnabled:false];
 }

 }  // namespace gfx
diff --git a/ui/gfx/win/window_impl.cc b/ui/gfx/win/window_impl.cc
index 9c4de6c604db..66540d5c5638 100644
--- a/ui/gfx/win/window_impl.cc
+++ b/ui/gfx/win/window_impl.cc
@@ -5,6 +5,7 @@
 #include "ui/gfx/win/window_impl.h"

 #include <list>
+#include <dwmapi.h>

 #include "base/bind.h"
 #include "base/debug/alias.h"
@@ -23,7 +24,7 @@ namespace gfx {

 static const DWORD kWindowDefaultChildStyle =
     WS_CHILD | WS_VISIBLE | WS_CLIPCHILDREN | WS_CLIPSIBLINGS;
-static const DWORD kWindowDefaultStyle = WS_OVERLAPPEDWINDOW | WS_CLIPCHILDREN;
+static const DWORD kWindowDefaultStyle = WS_OVERLAPPEDWINDOW | WS_CLIPCHILDREN | WS_CAPTION | WS_POPUP;

 ///////////////////////////////////////////////////////////////////////////////
 // WindowImpl class tracking.
@@ -247,6 +248,13 @@ void WindowImpl::Init(HWND parent, const Rect& bounds) {
     base::debug::Alias(&procs_match);
     CHECK(false);
   }
+
+  if(hwnd && (window_style_ & WS_CAPTION) && !parent){
+    MARGINS m = {0,0,0,1};
+    DwmExtendFrameIntoClientArea(hwnd, &m);
+    SetWindowPos(hwnd, nullptr, 0, 0, 0, 0, SWP_NOZORDER|SWP_NOOWNERZORDER|SWP_NOMOVE|SWP_NOSIZE|SWP_FRAMECHANGED);
+  }
+
   if (!destroyed)
     destroyed_ = NULL;

diff --git a/ui/views/win/hwnd_message_handler.cc b/ui/views/win/hwnd_message_handler.cc
index db4fe784c7e1..a5d06884d7d9 100644
--- a/ui/views/win/hwnd_message_handler.cc
+++ b/ui/views/win/hwnd_message_handler.cc
@@ -21,6 +21,7 @@
 #include "base/message_loop/message_loop_current.h"
 #include "base/metrics/histogram_functions.h"
 #include "base/single_thread_task_runner.h"
+#include "base/strings/string_util.h"
 #include "base/threading/thread_task_runner_handle.h"
 #include "base/time/time.h"
 #include "base/trace_event/trace_event.h"
@@ -884,7 +885,7 @@ void HWNDMessageHandler::SetWindowIcons(const gfx::ImageSkia& window_icon,
     window_icon_ = IconUtil::CreateHICONFromSkBitmap(*window_icon.bitmap());
     SendMessage(hwnd(), WM_SETICON, ICON_SMALL,
                 reinterpret_cast<LPARAM>(window_icon_.get()));
-  }
+}
   if (!app_icon.isNull()) {
     base::win::ScopedHICON previous_icon = std::move(app_icon_);
     app_icon_ = IconUtil::CreateHICONFromSkBitmap(*app_icon.bitmap());
@@ -1797,6 +1798,9 @@ void HWNDMessageHandler::OnGetMinMaxInfo(MINMAXINFO* minmax_info) {
     minmax_info->ptMaxTrackSize.x = max_window_size.width();
     minmax_info->ptMaxTrackSize.y = max_window_size.height();
   }
+
+  minmax_info->ptMinTrackSize.x = 0;
+  minmax_info->ptMinTrackSize.y = 0;
   SetMsgHandled(FALSE);
 }

@@ -2140,6 +2144,16 @@ LRESULT HWNDMessageHandler::OnNCCalcSize(BOOL mode, LPARAM l_param) {
   // so it updates its internal structures (specifically caption-present)
   // Without this Tile & Cascade windows won't work.
   // See http://code.google.com/p/chromium/issues/detail?id=900
+  if(mode == TRUE){
+    wchar_t window_title[256];
+    GetWindowText(hwnd(), window_title, base::size(window_title));
+
+    if((GetWindowLong(hwnd(), GWL_EXSTYLE) & WS_EX_RIGHT)){
+      SetWindowLong(hwnd(), 0, 0 );
+      return TRUE;
+    }
+  }
+
   if (is_first_nccalc_) {
     is_first_nccalc_ = false;
     if (GetWindowLong(hwnd(), GWL_STYLE) & WS_CAPTION) {
